packages/ui/src/views/w3m-wallet-filter-view/index.ts

```diff
@@ -0,0 +1,73 @@
+import type { DesktopConnectorData } from '@web3modal/core'
+import { CoreHelpers, RouterCtrl } from '@web3modal/core'
+import { html, LitElement } from 'lit'
+import { customElement, state } from 'lit/decorators.js'
+import '../../components/w3m-modal-content'
+import '../../components/w3m-modal-header'
+import '../../components/w3m-search-input'
+import '../../components/w3m-wallet-button'
+import { global } from '../../utils/Theme'
+import { getCustomWallets, handleMobileLinking } from '../../utils/UiHelpers'
+import styles, { dynamicStyles } from './styles'
+
+@customElement('w3m-wallet-filter-view')
+export class W3mWalletFilterView extends LitElement {
+  public static styles = [global, styles]
+
+  // -- state & properties ------------------------------------------- //
+  @state() private search = ''
+
+  // -- private ------------------------------------------------------ //
+  private async onConnectPlatform({ name, universal, native, walletId }: DesktopConnectorData) {
+    if (CoreHelpers.isMobile()) {
+      await handleMobileLinking({ native, universal }, name)
+    } else {
+      RouterCtrl.push('DesktopConnector', {
+        DesktopConnector: { name, walletId, universal, native }
+      })
+    }
+  }
+
+  private onSearchChange(event: Event) {
+    const { value } = event.target as HTMLInputElement
+    this.search = value
+  }
+
+  // -- render ------------------------------------------------------- //
+  protected render() {
+    const wallets = getCustomWallets()
+    const filtered = this.search.length
+      ? wallets.filter(wallet => wallet.name.toUpperCase().includes(this.search.toUpperCase()))
+      : wallets
+
+    return html`
+      ${dynamicStyles()}
+
+      <w3m-modal-header>
+        <w3m-search-input .onChange=${this.onSearchChange.bind(this)}></w3m-search-input>
+      </w3m-modal-header>
+
+      <w3m-modal-content>
```

Comment:
Minor: Curious about the way the CSS / transition works for the `Modal Content.` Is it preset to a certain height because I can imagine a lot of custom wallets not reaching 25+ explorer content.

I.e. the example in `custom.tsx,` the modal jumps height (under the assumption we have 25+?) + provides a lot of empty space. 

![image](https://user-images.githubusercontent.com/45455218/203518926-1cf9bbbd-27e8-4ff7-b834-02b4118d1b5a.png).

Could we adjust with a ternary that if less than ~20 wallets, no modal height change / jump? (Same Height as the previous page of `ConnectWalletView`?

Reply:
Good catch, there was a need for this in normal explorer view to avoid jumpy content during loading, but since all wallets are known in custom one, height can be dynamic 👍 updated both to feel nicer.